### PR TITLE
feat: update to use RectorConfig class

### PR DESCRIPTION
**Pull request by owenvoke**
Related to JIRA-5515 👍🏻

### CODE REVIEW

**Review by olivernybroe — COMMENTED** [review]
Looks pretty good, however it seems like some of our phpstan rules are breaking in the new version 🤔

**Comment by owenvoke** [comment]
Yep, just in the process of fixing that. Think it's because we can no longer use the `parent` selector. But just checking.

**Comment by owenvoke** [comment]
Just rebased and force-pushed onto `develop` following the changes by Odinn. 👍🏻

**Comment by owenvoke** [comment]
@lukeraymonddowning, a temporary workaround for this is to apply the following patch. However, that causes PHPStan to be less memory-efficient. Hopefully we can resolve this by updating some other way. 👍🏻

```diff
diff --git a/larastan.neon b/larastan.neon
index 1aeb5a0..47d6347 100644
--- a/larastan.neon
+++ b/larastan.neon
@@ -1,5 +1,6 @@
 includes:
     - ./phpstan.neon
+    - ./phpstan-rich-parser.neon
     - ../../nunomaduro/larastan/extension.neon
 
 parameters:
diff --git a/phpstan-rich-parser.neon b/phpstan-rich-parser.neon
new file mode 100644
index 0000000..de14642
--- /dev/null
+++ b/phpstan-rich-parser.neon
@@ -0,0 +1,3 @@
+conditionalTags:
+  PhpParser\NodeVisitor\NodeConnectingVisitor:
+    phpstan.parser.richParserNodeVisitor: true
diff --git a/tests/PHPStan/BaseRuleTestCase.php b/tests/PHPStan/BaseRuleTestCase.php
index e92b4ed..5d710f0 100644
--- a/tests/PHPStan/BaseRuleTestCase.php
+++ b/tests/PHPStan/BaseRuleTestCase.php
@@ -13,4 +13,11 @@ class BaseRuleTestCase extends RuleTestCase
     {
         return $this->rule;
     }
+
+    public static function getAdditionalConfigFiles() : array
+    {
+        return array_merge(parent::getAdditionalConfigFiles(), [
+            __DIR__ . '/../../phpstan-rich-parser.neon',
+        ]);
+    }
 }
\ No newline at end of file
```

**Comment by olivernybroe** [comment]
@owenvoke maybe we should do that for now and then when Luke has time to update the rule we can upgrade to the new memory efficient edition

**Comment by owenvoke** [comment]
@olivernybroe, sure! Sounds like a plan for now. 🙂

**Review by olivernybroe — APPROVED** [review]

File: src/WorksomeRectorConfig.php [hunk 1]
```diff
@@ -41,8 +42,8 @@ public static function setup(RectorConfig $rectorConfig): void
         $rectorConfig->rule(RemoveUnusedVariableInCatchRector::class);
 
         // DEAD CODE set rules
-        $rectorConfig->rule(\Rector\DeadCode\Rector\FunctionLike\RemoveCodeAfterReturnRector::class);
-        $rectorConfig->rule(\Rector\DeadCode\Rector\ClassMethod\RemoveDeadConstructorRector::class);
+        $rectorConfig->rule(RemoveUnreachableStatementRector::class);
```
Comment (owenvoke) [inline src/WorksomeRectorConfig.php:45]: The `Rector\DeadCode\Rector\FunctionLike\RemoveCodeAfterReturnRector` rule was removed in https://github.com/rectorphp/rector-src/pull/2199, so I've replaced it with `RemoveUnreachableStatementRector` which does the same and more. 👍🏻

**Pull request merged**